src/main/java/com/minecolonies/coremod/tileentities/TileEntityBakerBuilding.java

```diff
@@ -0,0 +1,77 @@
+package com.minecolonies.coremod.tileentities;
+
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.colony.buildings.BuildingBaker;
+import com.minecolonies.coremod.entity.ai.citizen.baker.BakerEnums;
+import com.minecolonies.coremod.entity.ai.citizen.baker.BakingProduct;
+import net.minecraft.block.BlockFurnace;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.init.Blocks;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Map;
+
+/**
+ * Baker building special TileEntity.
+ */
+public class TileEntityBakerBuilding extends TileEntityColonyBuilding
+{
+    /**
+     * Wait this amount of ticks before checking again.
+     */
+    private static final int WAIT_TICKS = 160;
+
+    /**
+     * Ticks past since the last check.
+     */
+    private int ticksPassed = 0;
+
+    @Override
+    public void update()
```

Comment:
Good point, just remembered that we have a way to tick the colony buildings as well.